plugins/pipe_text.py

```diff
@@ -0,0 +1,59 @@
+import sublime
+import sublime_plugin
+from subprocess import run, PIPE
+import time
+from datetime import datetime
+
+
+class PipeTextCommand(sublime_plugin.TextCommand):
+    """Pipe text from ST - the selections, if any, otherwise the entire buffer contents
+       - to the specified shell command.
+       Useful for formatting XML or JSON in a quick and easy manner.
+       i.e. a workaround for https://github.com/sublimehq/sublime_text/issues/3294
+    """
+    def run(self, edit, shell_command):
+        if all((region.empty() for region in self.view.sel())):
+            regions = [sublime.Region(0, self.view.size())]
+        else:
+            regions = self.view.sel()
+
+        failures = False
+        start = time.perf_counter()
+        logs = list()
+        def log(message):
+            nonlocal logs
+            log_text = str(datetime.now()) + ' ' + message
+            logs.append(log_text)
+            print(log_text)
+
+        for region in reversed(regions):
```

Comment:
Technically not required to reverse here, but doesn't matter as it will be more robust, should the selections be expanded to lines above, like I pondered.